### PR TITLE
[Doc] Update release notes

### PR DESCRIPTION
**Pull request by jcalcaben**
## Description

Updates the release notes with information about the home page.

## Related Issue

Follow up for #2565 

## Acceptance 

@dpatil-magento 

### Verification Stakeholders

@dpatil-magento 

### Verification Steps

1. Open CHANGELOG.md
1. Verify content is accurate

## Checklist

- [ ] I have added tests to cover my changes, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Comment by devops-pwa-codebuild** [comment]
**Performance Test Results**

The following fails have been reported by WebpageTest. These numbers indicates a possible performance issue with the PR which requires further manual testing to validate.

https://pr-2568.pwa-venia.com : LH Performance Expected 0.85 Actual 0.57, LH Best Practices Expected 1 Actual 0.92
https://pr-2568.pwa-venia.com/venia-tops.html : LH Performance Expected 0.75 Actual 0.36, LH Best Practices Expected 1 Actual 0.92
https://pr-2568.pwa-venia.com/valeria-two-layer-tank.html : LH Performance Expected 0.8 Actual 0.49, LH Accessibility Expected 0.9 Actual 0.89, LH Best Practices Expected 1 Actual 0.92

File: CHANGELOG.md [hunk 1]
```diff
@@ -70,7 +70,7 @@ In addition to the style changes for Venia, release 7.0.0 also adds a new home p
 The content of this CMS page is defined in the Magento Admin using the Page Builder extension.
 It showcases Page Builder content types such as Sliders, Banners, and Gallery Items.
 
-Check it out: https://develop.pwa-venia.com/venia-new-home
+This new page replaces the old home page content in Venia.
```
Comment (cherdman) [inline CHANGELOG.md:73]: We should provide a hotlink to the new page.

Comment (jcalcaben) [reply CHANGELOG.md:73]: We can provide a link to the 7.0 staging server but that is only valid until the release. Post release, we will have to change it back to the develop link.

Comment (jcalcaben) [reply CHANGELOG.md:73]: Resolve by: 98b269c

**Pull request merged**